### PR TITLE
Fixing dev mode

### LINKED ISSUE #1063: --dev flag throws seg fault
### Expected Behavior

Starting upstream (i.e. Ethereum's ) geth with --dev flag would normally start up a local network with the clique consensus engine. Since we no longer have clique in celo-blockchain, using --dev throws a seg fault. Expected behavior would be for --dev to start up succesfully with istanbul. 

### Current Behavior

--dev seg faults because the genesis block it uses is incompatible with istanbul (see DeveloperGenesisBlock in genesis.go). We've received feedback from an external partner that adding back an istanbul compatible version of dev mode would be very helpful. 

### Steps to Reproduce Behavior

`geth --dev --nodiscover --maxpeers 0 --light.maxpeers 0 --mine --minerthreads 1 --datadir /data --verbosity 3 --rpc --rpcaddr 0.0.0.0 --rpcapi eth,net --nousb`

### Logs

`WARN [05-30|00:20:29.778] Error getting snapshot                   number=0 hash=c98bf6…d25f28 err="rlp: expected input list for struct { AddedValidators []common.Address; AddedValidatorsPublicKeys []blscrypto.SerializedPublicKey; RemovedValidators *big.Int; Seal []uint8; AggregatedSeal types.IstanbulAggregatedSeal; ParentAggregatedSeal types.IstanbulAggregatedSeal }"
ERROR[05-30|00:20:29.779] Unable to extract istanbul extra         err="rlp: expected input list for struct { AddedValidators []common.Address; AddedValidatorsPublicKeys []blscrypto.SerializedPublicKey; RemovedValidators *big.Int; Seal []uint8; AggregatedSeal types.IstanbulAggregatedSeal; ParentAggregatedSeal types.IstanbulAggregatedSeal }"
WARN [05-30|00:20:29.779] Error getting snapshot                   number=0 hash=c98bf6…d25f28 err="rlp: expected input list for struct { AddedValidators []common.Address; AddedValidatorsPublicKeys []blscrypto.SerializedPublicKey; RemovedValidators *big.Int; Seal []uint8; AggregatedSeal types.IstanbulAggregatedSeal; ParentAggregatedSeal types.IstanbulAggregatedSeal }"
INFO [05-30|00:20:29.779] Commit new mining work                   number=1 sealhash=197825…04291d txs=0 gas=0 fees=0 elapsed=294.459µs
ERROR[05-30|00:20:29.779] Unable to extract istanbul extra         err="rlp: expected input list for struct { AddedValidators []common.Address; AddedValidatorsPublicKeys []blscrypto.SerializedPublicKey; RemovedValidators *big.Int; Seal []uint8; AggregatedSeal types.IstanbulAggregatedSeal; ParentAggregatedSeal types.IstanbulAggregatedSeal }"
WARN [05-30|00:20:29.779] Block sealing failed                     err="rlp: expected input list for struct { AddedValidators []common.Address; AddedValidatorsPublicKeys []blscrypto.SerializedPublicKey; RemovedValidators *big.Int; Seal []uint8; AggregatedSeal types.IstanbulAggregatedSeal; ParentAggregatedSeal types.IstanbulAggregatedSeal }"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xc98620]

goroutine 252 [running]:
github.com/ethereum/go-ethereum/consensus/istanbul/core.(*roundStateImpl).EncodeRLP(0xc000bb4100, 0x1882100, 0xc000c3c050, 0x0, 0x0)
    /opt/testnet-celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/consensus/istanbul/core/roundstate.go:528 +0x100
github.com/ethereum/go-ethereum/rlp.makeEncoderWriter.func2(0x15bd9c0, 0xc000bb4100, 0x199, 0xc000c3c050, 0xc000bb4100, 0x199)
    /opt/testnet-celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/rlp/encode.go:541 +0x15a
github.com/ethereum/go-ethereum/rlp.makePtrWriter.func1(0x15c71e0, 0xc000bb4100, 0x16, 0xc000c3c050, 0x15c0000, 0xc000b9e060)
    /opt/testnet-celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/rlp/encode.go:523 +0xba
github.com/ethereum/go-ethereum/rlp.(*encbuf).encode(0xc000c3c050, 0x15c71e0, 0xc000bb4100, 0x4b222c, 0x20)`

### System information

Run `geth version`

### PR DESCRIPTION
**Pull request by mrsmkl**
### Description

Dev mode now works. As the validator it uses the address `0x47e172F6CfB6c7D01C1574fa3E2Be7CC73269D95` from monorepo.

### Tested

Only manual testing.

### Related issues

- Fixes #1063

### Backwards compatibility

Yes.

### CODE REVIEW

File: core/genesis.go [hunk 1]
```diff
@@ -383,21 +382,12 @@ func DefaultAlfajoresGenesisBlock() *Genesis {
...
-			common.BytesToAddress([]byte{7}): {Balance: big.NewInt(1)}, // ECScalarMul
-			common.BytesToAddress([]byte{8}): {Balance: big.NewInt(1)}, // ECPairing
-			faucet:                           {Balance: new(big.Int).Sub(new(big.Int).Lsh(big.NewInt(1), 256), big.NewInt(9))},
-		},
+		ExtraData: hexutil.MustDecode("0xecc833a7747eaa8327335e8e0c6b6d8aa3a38d0063591e43ce116ccf5c89753ef9018ad59447e172f6cfb6c7d01c1574fa3e2be7cc73269d95f862b8604fa3f67fc913878b068d1fa1cdddc54913d3bf988dbe5a36a20fa888f20d4894c408a6773f3d7bde11154f2a3076b700d345a42fd25a0e5e83f4db5586ac7979ac2053cd95d8f2efd3e959571ceccaa743e02cf4be3f5d7aaddb0b06fc9aff0080b8410000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000f86480b86000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080f86480b86000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080"),
```
Comment (mcortesi) [inline core/genesis.go:390]: can we move this and other `extraData` dumps into a new file as constants? They are huge, almost break editors

File: params/config.go [hunk 2]
```diff
@@ -137,8 +137,10 @@ var (
 	}
 
 	DefaultChainConfig = &ChainConfig{big.NewInt(1337), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, &IstanbulConfig{
```
Comment (mcortesi) [inline params/config.go:139]: I think the defauls should be `mainnet` or non existent.

If not, we risk using a dev config on a mainnet node if we missplace a config.

Comment (mrsmkl) [reply params/config.go]: Checked that mainnet is default, and changed the name of this config to `IstanbulTestConfig`.

File: core/genesis.go (hunk 1 above)
Comment (mrsmkl) [reply core/genesis.go]: Yeah, it takes 10 seconds to type a character to those lines.

**Pull request merged**